src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// Taken from https://github.com/dotnet/cli/blob/8f7b58dd665f60300221a4eb5910a691749396aa/src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs
+
+#if NETSTANDARD2_0
+using System.Diagnostics;
+
+namespace Microsoft.Build.Locator
+{
+
+    using System;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    internal static class DotNetSdkLocationHelper
+    {
+        public static VisualStudioInstance GetInstance(string workingDir)
+        {
+            string dotNetSdkPath = hostfxr_resolve_sdk(workingDir);
+
+            if (String.IsNullOrWhiteSpace(dotNetSdkPath))
+            {
+                return null;
+            }
+
+            if (!File.Exists(Path.Combine(dotNetSdkPath, "Microsoft.Build.dll")))
+            {
+                return null;
+            }
+
+            FileInfo dotnetAssemblyPath = new FileInfo(Path.Combine(dotNetSdkPath, "dotnet.dll"));
+
+            if (!dotnetAssemblyPath.Exists)
+            {
+                return null;
+            }
+
+            FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(dotnetAssemblyPath.FullName);
+
+            return new VisualStudioInstance(
+                name: ".NET Core SDK",
+                path: dotNetSdkPath,
+                version: new Version(fileVersionInfo.FileMajorPart, fileVersionInfo.FileMinorPart, fileVersionInfo.FileBuildPart),
+                discoveryType: DiscoveryType.DotNetSdk);
+        }
+
+        private static string hostfxr_resolve_sdk(string workingDir)
+        {
+            Process process = Process.GetCurrentProcess();
```

Comment:
Oh. Maybe I'm wrong about using `realpath` on *nix if this is how we're getting `dotnet`. I can see how this might work for shared framework apps, but how would it work for standalone .NET Core apps?

Reply:
Yeah it should work as its just a copy/paste.  I don't think standalone apps need MSBuildLocator since they'd need to have the full closure of everything needed in order to use MSBuild.  They'd have to reference a lot more NuGet packages like compilers etc.  I wonder if we can detect that its a standalone app and error out?  Do you know of any way to distinguish at run time?

Reply:
No idea. I guess you're right that standalone apps are really out of luck. In that case, I suppose they need to basically include everything needed to run `dotnet build` themselves, including MSBuild and the SDK resolvers.